app/src/main/java/com/example/internetshop/presentation/activity/MainActivity.kt

```diff
@@ -133,4 +140,21 @@ class MainActivity : AppCompatActivity(), ContainerHolder {
     override fun getContainerId(): Int? {
         return binding?.fragmentContainer?.id
     }
+
+    private fun updateFavoriteDate() {
+        serviceState.event.observe(this, {
+            when (it) {
+                is UpdateFavoriteProductDataServiceState
```

Comment:
Please rename `UpdateFavoriteProductDataServiceState` into `FetchFavoritesServiceState`. And also rename `UpdateFavoriteProductDataServiceEvent` into `Event`. Because when you call event there is tautology. After renaming you will be able to call event by `FetchFavoritesServiceState.Event.ServiceDestroying`. This will reduce amount of text and make code more readable and understandable.

Reply:
fixed